tests/PHPUnit/Unit/DbHelperTest.php

```diff
@@ -21,6 +21,56 @@
  */
 class DbHelperTest extends \PHPUnit\Framework\TestCase
 {
+    /**
+     * @dataProvider getExtractOrderByFromQueryTestData
+     */
+    public function testExtractOrderByFromQuery(string $sql, ?string $expectedOrderBy): void
+    {
+        $extractedOrderBy = DbHelper::extractOrderByFromQuery($sql);
+
+        // compare with collapsed whitespace
+        $expectedOrderBy = trim(preg_replace('/\s+/', ' ', $expectedOrderBy));
+        $extractedOrderBy = trim(preg_replace('/\s+/', ' ', $extractedOrderBy));
+
+        $this->assertSame($expectedOrderBy, $extractedOrderBy);
+    }
+
+    public function getExtractOrderByFromQueryTestData(): iterable
```

Comment:
Should we maybe add a test with nested queries? What is the behaviour meant to be there like, should it only extract from the outer query?

Reply:
A nested clause would make the parentheses check fail, and return nothing. But it would also fail with multiple ORDER BYs (one inner + one outer).

Having just an inner one should always be caught by the parentheses check, and having multiple clauses is only a minor regex adjustment away :+1: